src/web.ml

```diff
@@ -364,15 +364,36 @@ module Make (R : Renderer) = struct
     in
     (src, StringMap.find pipeline_id pipelines)
 
-  let show_pipeline ~(state : t) pipeline_source_id pipeline_id =
+  let show_pipeline ctx ~(state : t) pipeline_source_id pipeline_id =
     let src, pipeline = find_pipeline ~state pipeline_source_id pipeline_id in
     let { user_meta; creation_date; _ } = pipeline.metadata in
     let date = Unix.gmtime creation_date in
     let datestr (date : Unix.tm) =
       Fmt.str "%02d/%02d/%4d %02d:%02d" date.tm_mday (1 + date.tm_mon)
         (1900 + date.tm_year) date.tm_hour date.tm_min
     in
+    let rebuildable_job_ids =
+      Jobs.rebuildable_jobs node_map_status pipeline.stages
+    in
+    let csrf = Current_web.Context.csrf ctx in
     let open Tyxml_html in
+    let rebuild_button =
+      if List.length rebuildable_job_ids == 0 then []
```

Comment:
`==` in OCaml is physical equality, you want to use `=` instead.
Don't use `List.length` to check if a list is empty, because it loops through all the list. The better way is to pattern-match the list, so `if rebuildable_jobs_ids = [] then []`.